sys/src/9/ufs/ffs/ffs_vfsops.c

```diff
@@ -773,13 +790,16 @@ ffs_mountfs (struct vnode *devvp, struct mount *mp, struct thread *td)
 		    fs->fs_bsize <= MAXBSIZE &&
 		    fs->fs_bsize >= sizeof(struct fs))
 			break;
-		brelse(bp);
-		bp = nil;
 	}
+
 	if (sblock_try[i] == -1) {
-		error = EINVAL;		/* XXX needs translation */
+		//error = EINVAL;		/* XXX needs translation */
+		// TODO HARVEY error string?
+		error = -1;
```

Comment:
should be using error() here but we can always fix that up later. The goto stuff is not really how it should be done but I realize this comes from unix.